### DISCUSSION TITLE
Welcome to ctrlr Discussions!

### DISCUSSION THREAD

**Discussion by RomanKubiak** [RomanKubiak/ctrlr#93] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tomlem** [comment]
Hi Roman,
May ctrlr be of great help for many musicians to deploy their arts for the benefit of all!

**Comment by dobo365** [comment]
Even if I said it already many times, thx Roman for having created Ctrlr that I have a lot of fun creating panels with and that takes most of my hobby time.

I'm _goodweather_ on the forum and after discovering Ctrlr a few years ago I'm building panels as hobby.
I developed a Step by Step guide to help newcomers building their own panels. I continued extending it with Lua and many other infos but this extension is not ready to be published as it needs quite some cleanup first.

Would be pleased to get some guidance/help on how to contribute to the coding with Visual Studio.
I can program, I can learn many things but miss the initial step by step support to start using Visual Studio with Ctrlr through Github.
Would like to know how to build Ctrlr, manage versions, propose changes, know what to add (libs), how it interacts with Juce, VST2/VST3 building, get panels for Ubuntu Studio on Linux,...
But first the basics ;-) How to build Ctrlr 5.3.201 on Visual Studio and how to propose, develop, test changes.
I posted 2 discussions thread about that.

**Comment by Possemo** [comment]
I see that the ctrlr.org forum does make problems (bugs, spam maybe phishing etc.). But there is a tremendous amount of information available in the threads built during all these years. As there is no real tutorial for ctrlr I think it would be a big loss if you nuke ctrlr.org. It would be nice if you could preserve the information somehow.
Best, and happy new year!
Chris

**Reply by RomanKubiak** [reply]
i'm not removing the page, the forums stay there just in read-only mode.

**Reply by re-syn** [reply]
Hi-
I'm resyn on ctrlr, cannot seem to log-in at all, and get no reminders from wordpress. Is the read-only status of the site implemented this way, ie no logins possible on the ctrlr wordpress site? Any any event: thanx for your work and for managing the move to github. CTRLR is one of the most valuable tools in computer music these days!

**Comment by shooking** [comment]
Hi @dobo365 @Possemo et al.
So I made some YT recently on how to build Ctrlr on Raspberry Pi. I added steps in the descriptions. Since then I build Dexed - and this forced me to build Projucer on Raspberry Pi - and now I can produce proper makefiles for Ctrlr on RPi - no hacking etc.

I see also Projucer produces the various VS solutions.
Likely I will create another one related to Pi - how to "properly" use the Projucer. Now that I get Ctrlr is a Projucer app - well a load of things become a lot clearer - most importantly that sometimes the bug lives in JUCE not in Ctrlr :-).

Regarding your specific questions

1. how to build Ctrlr: I have a YT on this for Pi4. Linux is nearly identical - less effort (just trying it now on Ubuntu). How to build an old timestamp - and soon will make how to build nightly. When I did this in the past I got some 6.x branch which was cool BUT there were a lot of changed to MemoryBlocks etc. It seems nightly build 5.4.1 now. And my YT shows how to get back to an older stable point.
2. manage versions - so I think you mean branching etc? This is done in the git and I figure Atom should be the one to make the branching - you can git pull, make suggestions then push for his check/merge or rejection.
3. propose changes - as above
4. know what to add (libs) - JUCE is your friend here - I am gonna make a YT on this because I am geeking out about it quietly to myself.
5. how it interacts with Juce - yeah I wasnt getting this at all ... now I see it IS a JUCE app. So Jules provides a framework (in the sourcefiles -   The Introjucer is part of the JUCE library - "Jules' Utility Class Extensions")
6. VST2/VST3 building - just VST3... vst2 are no longer supported. And great news .. this more or less comes for free out of JUCE.
7. get panels for Ubuntu Studio on Linux - not sure what this means. Oh it is a Studio APP - nice. I used to use CCRMA but for sure going to check this out. Great tip.
Building Windows - I loved MS VS of old. But this thing is bloatware now. Worse than the PS4 experience - insert bluray .. wait 2 days for 25GB to download. I can get you productive in Linux in no time - there again you likely already are.

**Reply by Possemo** [reply]
Hey that sounds great. I think Ctrlr for Raspberry could be the fundament for cool cheap-DIY-hardware projects.

**Reply by dobo365** [reply]
Thx for your hints and answers, @shooking 
As I said above, I'm not specifically looking for Linux. First target is to simply build it for classical Windows.
Will follow first the instructions provided by tedjuh for VS2010 and 5.3.201 then will look to VS2019.
Need also to test the latest 5.5.x version made by Roman (whereof 5.5.1 on Linux).

**Comment by Ted-juh** [comment]
The manual I made is for VS2010. My plan is to see if I can transfer the project to VS2019 this weekend.

**Reply by shooking** [reply]
hi @Tedjuh1976 
I dont think any platform builds "stock" from a vanilla git pull.

I have tried Linux on Ubuntu.
Raspberry Pi.
MS VS 2019.
Turn of the year things built - so it seems like the master branch is not designed to be clean?
JUCE on the other hand builds clean. And has tags/releases to version builds.

If you discover the necessary git checkout points for Ctrlr please let us know how you did it.

There are missing files etc.
For example:

c:\Pilots\ctrlr\Source\UIComponents\CtrlrPanel\CtrlrPanelProperties.h(3,10): fatal error C1083: Cannot open include file: 'CtrlrPanelFindProperty.h': No such file or directory (compiling source file ..\..\Source\UICompone
nts\CtrlrComponents\Sliders\CtrlrImageSlider.cpp) [c:\Pilots\ctrlr\Builds\VisualStudio2019\Ctrlr_SharedCode.vcxproj]

c:\Pilots\ctrlr>dir /s CtrlrPanelFindProperty.h
 Volume in drive C is OSDisk
 Volume Serial Number is E8CE-E884
File Not Found

So why you look for it Steve?


c:\Pilots\ctrlr>find /i "CtrlrPanelFindProperty.h" Ctrlr.jucer

---------- CTRLR.JUCER
          <FILE id="DP5ROh" name="CtrlrPanelFindProperty.h" compile="0" resource="0"
                file="Source/UIComponents/CtrlrPanel/CtrlrPanelFindProperty.h"/>

so as it stands the branch doesnt build vanilla.
What tricks are required?

In the past I have had to 
cd ctrlr
git checkout 3d8651d991d377f7b167806c39c47a3121fe9536
Archangel gave this hint


c:\Pilots\ctrlr>git fetch --all --tags
Fetching origin

so I dont see any tags on the branch?

c:\Pilots\ctrlr>git branch
* master

so I am in master.

I can search thru git log but do not see an obvious commit point to checkout.
As I said the previous 5.x branch I used was
git checkout 3d8651d991d377f7b167806c39c47a3121fe9536

Any hints on checking the code out at known good build points are much appreciated.

**Comment by damiensellier** [comment]
Hi guys, I'm Damien from the Ctrlr Forum. Happy everybody's slowly moving to this github feed.
I don't really know how github works, but if possible you probably should split the feeds by categories like on the forum (Ctrlr software bug reports, Pannel coding, etc)
Thanks for everything, Ctrlr is awesome, much love to the community as well :-)
Damien

**Comment by Godlike-Productions** [comment]
Good to see so many of the old gang over here. I'm neck deep in some very complex panels at the moment, so won't be testing new versions for the time being.
